FinniversKit/Sources/Components/MonthAndYearPickerView/MonthAndYearPickerView.swift

```diff
@@ -0,0 +1,190 @@
+import UIKit
+
+public class MonthAndYearPickerView: UIControl {
+    public var locale: Locale = .autoupdatingCurrent {
+        didSet {
+            calendar.locale = locale
+        }
+    }
+
+    public var timeZone: TimeZone = .autoupdatingCurrent {
+        didSet {
+            calendar.timeZone = timeZone
+        }
+    }
+
+    public var selectedDate: Date = Date() {
+        didSet {
+            guard selectedDate != oldValue else { return }
+
+            if let maximumDate = maximumDate, selectedDateIsAfter(maximumDate) {
+                selectedDate = maximumDate
+                setPickerViewSelection(selectedDate, animated: true)
+            } else if let minimumDate = minimumDate, selectedDateIsBefore(minimumDate) {
+                selectedDate = minimumDate
+                setPickerViewSelection(selectedDate, animated: true)
+            }
+
+            sendActions(for: .valueChanged)
+        }
+    }
+
+    public var maximumDate: Date? {
+        didSet {
+            if let maximumDate = maximumDate, selectedDateIsAfter(maximumDate) {
+                selectedDate = maximumDate
+            }
+        }
+    }
+
+    public var minimumDate: Date? {
+        didSet {
+            if let minimumDate = minimumDate, selectedDateIsBefore(minimumDate) {
+                selectedDate = minimumDate
+            }
+        }
+    }
+
+    public let minimumYear: Int = 1900
+
+    private var calendar = Calendar(identifier: .iso8601)
+
+    private var years: [Int] {
+        let currentYear = calendar.component(.year, from: Date())
+        return [Int](minimumYear...currentYear)
+    }
+
+    private var months: [String] {
+        calendar.monthSymbols
+    }
+
+    private lazy var pickerView: UIPickerView = {
+        let view = UIPickerView(withAutoLayout: true)
+        view.dataSource = self
+        view.delegate = self
+        return view
+    }()
+
+    public convenience init() {
+        self.init(frame: .zero)
+    }
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setup() {
+        calendar.locale = locale
+        calendar.timeZone = timeZone
```

Comment:
These won't be set if the `init()` is used, since `setup()` isn't called from there. Maybe `calendar` should be refactored into a lazy var instead and move these there?

Reply:
Same with the lines adding and filling `pickerView` as a subview below. Do we need the `init()` at all?

Reply:
`setup` is called. The convenience initializer calls `self.init(frame:)`, not `super.init(frame:)` which in turn calls `setup` :)

Reply:
My bad, didn't notice it was calling `self`😄